pkg/redis/client.go

```diff
@@ -496,6 +642,28 @@ func (c *redisClient) ZRevRange(ctx context.Context, key string, start int64, st
 	return cmd.(*baseRedis.StringSliceCmd).Val(), err
 }
 
+func (c *redisClient) ZRevRangeWithScores(ctx context.Context, key string, start int64, stop int64) ([]Z, error) {
```

Comment:
It seems that this is a deprecated function, so we should not add it. See: https://redis.io/commands/zrevrange/

Quote:
```
As of Redis version 6.2.0, this command is regarded as deprecated.

It can be replaced by [ZRANGE](https://redis.io/commands/zrange) with the REV argument when migrating or writing new code.
```

I think you might be able to achieve the same with `ZRangeArgsWithScores` from the base client.